jetty-server/src/main/java/org/eclipse/jetty/server/handler/InetAccessSet.java

```diff
@@ -0,0 +1,159 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server.handler;
+
+import java.net.InetAddress;
+import java.util.AbstractSet;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.eclipse.jetty.http.pathmap.PathSpec;
+import org.eclipse.jetty.http.pathmap.ServletPathSpec;
+import org.eclipse.jetty.util.InetAddressPattern;
+import org.eclipse.jetty.util.StringUtil;
+
+public class InetAccessSet extends AbstractSet<InetAccessSet.PatternTuple> implements Set<InetAccessSet.PatternTuple>, Predicate<InetAccessSet.AccessTuple>
+{
+    private ArrayList<PatternTuple> tuples = new ArrayList<>();
+
+    @Override
+    public boolean add(PatternTuple storageTuple)
+    {
+        return tuples.add(storageTuple);
+    }
+
+    @Override
+    public boolean remove(Object o)
+    {
+        return tuples.remove(o);
+    }
+
+    @Override
+    public Iterator<PatternTuple> iterator()
+    {
+        return tuples.iterator();
+    }
+
+    @Override
+    public int size()
+    {
+        return tuples.size();
+    }
+
+    @Override
+    public boolean test(AccessTuple entry)
+    {
+        if (entry == null)
+            return false;
+
+        // TODO: Can this be done more efficiently with something like PathMappings.
```

Comment:
I'm not sure we can do much better than this for the common case of just an address pattern.  `InetAddressSet` internally does an iteration anyway, so in the common case we are just the same.   So I'd probably remove the TODO.